### PR TITLE
Replacement of NewtonSoft.JSON by System.Text.Json

### PR DESCRIPTION
**Pull request by Adilson**
## Description
Replacement of NewtonSoft.JSON by System.Text.Json, assuming it´s a brand new implementation of JSON parsing and serialization with better benchmarks 

## Motivation and Context
It´ll reduce a exogenous dependency and improve performance since from .net core 3.1 onward it will use new Span<> faster structures.
This is addressing the issue #58, myself opened it.

## Testing
The test which evaluate indirectly this feature is working.

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to change)

It´s not exactly a breaking change, but since it changes the dependency list, it cares some attention to publish.

## Checklist

- [x] My code follows the code style of this project
- [ ] My change requires a change to the documentation
- [ ] I have updated the documentation accordingly
- [x] I have read the **README** document
- [ ] I have added tests to cover my changes
- [x] All new and existing tests passed

## License

- [x] I confirm that this pull request can be released under the Apache 2 license

[issues]: https://github.com/aws/aws-dotnet-extensions-configuration/issues
[license]: http://aws.amazon.com/apache2.0/
[cla]: http://en.wikipedia.org/wiki/Contributor_License_Agreement

### CODE REVIEW

File: src/Amazon.Extensions.Configuration.SystemsManager/Internal/JsonConfigurationParser.cs [hunk 1]
```diff
@@ -19,9 +19,11 @@
 using System.Globalization;
 using System.IO;
 using System.Linq;
+using System.Text;
+using System.Text.Json;
 using Microsoft.Extensions.Configuration;
-using Newtonsoft.Json;
-using Newtonsoft.Json.Linq;
+//using Newtonsoft.Json;
+//using Newtonsoft.Json.Linq;
```
Comment (rcollette) [inline src/Amazon.Extensions.Configuration.SystemsManager/Internal/JsonConfigurationParser.cs:26]: Remove commented code.

File: src/Amazon.Extensions.Configuration.SystemsManager/Internal/JsonConfigurationParser.cs [hunk 2]
```diff
@@ -33,92 +35,82 @@ public class JsonConfigurationParser : IDisposable
...
-            }
         }
 
-        private void VisitArray(JArray array)
+        private void VisitArray(JsonElement array)
         {
-            for (var index = 0; index < array.Count; index++)
-            {
+            int index = 0;
+            foreach (var item in array.EnumerateArray())
```
Comment (rcollette) [inline src/Amazon.Extensions.Configuration.SystemsManager/Internal/JsonConfigurationParser.cs:97]: I'm curious, why switch to foreach when you sill need an array index?

File: src/Amazon.Extensions.Configuration.SystemsManager/Internal/JsonConfigurationParser.cs [hunk 3]
```diff
@@ -33,92 +35,82 @@ public class JsonConfigurationParser : IDisposable
...
-            using (var reader = new StringReader(input))
-            using (var parser = new JsonConfigurationParser())
-            {
-                return parser.ParseInput(reader);    
-            }
+        public static IDictionary<string, string> Parse(string input)
+        {
+            using (var doc = JsonDocument.Parse(input))
+            {
+                using (var parser = new JsonConfigurationParser())
```
Comment (rcollette) [inline src/Amazon.Extensions.Configuration.SystemsManager/Internal/JsonConfigurationParser.cs:54]: Why is an additional nesting introduced?

File: src/Amazon.Extensions.Configuration.SystemsManager/Internal/JsonConfigurationParser.cs [hunk 4]
```diff
@@ -33,92 +35,82 @@ public class JsonConfigurationParser : IDisposable
...
-
         public static IDictionary<string, string> Parse(Stream input)
         {
-            using (var reader = new StreamReader(input))
-            using (var parser = new JsonConfigurationParser())
-            {
-                return parser.ParseInput(reader);    
+            using (var doc = JsonDocument.Parse(input))
+            {
+                using (var parser = new JsonConfigurationParser())
```
Comment (rcollette) [inline src/Amazon.Extensions.Configuration.SystemsManager/Internal/JsonConfigurationParser.cs:42]: Why is an additional nesting introduced?

**Comment by normj** [comment]
I have created PR [#103 ](https://github.com/aws/aws-dotnet-extensions-configuration/pull/103) that consolidates this open PRs along with this change. Because of the breaking changes across the PRs including this one I am planning on a major version bump to 4.

I'm going to close this PR and track the work in #103. Please feel free to review that PR for the changes I made from this PR.

**Pull request closed without merging**